openmc/settings.py

```diff
@@ -512,37 +493,15 @@ def set_source_space(self, stype, params):
 
         """
 
-        if not is_string(stype):
-            msg = 'Unable to set source space type to a non-string ' \
-                  'value {0}'.format(stype)
-            raise ValueError(msg)
-
-        elif stype not in ['box', 'fission', 'point']:
-            msg = 'Unable to set source space type to {0} since it is not ' \
-                  'box or point'.format(stype)
-            raise ValueError(msg)
-
-        elif not isinstance(params, (tuple, list, np.ndarray)):
-            msg = 'Unable to set source space parameters to {0} since it is ' \
-                  'not a Python tuple, list or NumPy array'.format(params)
-            raise ValueError(msg)
-
-        elif stype in ['box', 'fission'] and len(params) != 6:
-            msg = 'Unable to set source space parameters for a box/fission ' \
-                  'distribution to {0} since it does not contain 6 values'\
-                      .format(params)
-            raise ValueError(msg)
-
-        elif stype == 'point' and len(params) != 3:
-            msg = 'Unable to set source space parameters for a point to {0} ' \
-                  'since it does not contain 3 values'.format(params)
-            raise ValueError(msg)
-
-        for param in params:
-            if not is_integer(param) and not is_float(param):
-                msg = 'Unable to set source space parameters to {0} since it ' \
-                      'is not an integer or floating point value'.format(param)
-                raise ValueError(msg)
+        check_type('source space type', stype, basestring)
+        check_value('source space type', stype, ['box', 'fission', 'point'])
+        check_type('source space parameters', params, Iterable, Real)
+        if stype in ['box', 'fission'] and len(params) != 6:
```

Comment:
why does this conditional include ``len(params) != 6`` only to then call ``check_length(...)``?